### PR TITLE
Mb 12310 Convert mto_service_item and mto_agent tests

### PR DESCRIPTION
**Pull request by shimonacarvalho**
## Summary

Converts all remaining tests in `/pkg/services/mto_agent`  and `/pkg/services/mto_service_item` to use transactions.

[Pattern for server tests conversion](https://transcom.github.io/mymove-docs/docs/backend/testing/running-server-tests-inside-a-transaction/) explains more about the approach used.

## Setup to Run Your Code

A clean run of `make server_test` is sufficient.

## Verification Steps

- [x] No subtests are run with `suite.T().Run(...)` - they all use `suite.Run(...)`
- [x] There is no unjustified usage of `Truncate` in the tests
- [x] There is no unjustified usage of `suite.AppContextForTest().DB()` - instead `suite.DB()` is used directly
- [x] `Fatalf` is deprecated in favor of other assertions/checks
- [x] Go's `testing` package is only imported in the setup test file

### CODE REVIEW

File: pkg/services/mto_service_item/mto_service_item_updater_test.go [hunk 1]
```diff
@@ -122,14 +129,18 @@ func (suite *MTOServiceItemServiceSuite) TestMTOServiceItemUpdater() {
...
 	now := time.Now()
 
+	// setupTestData := func() (models.MTOServiceItem, models.MTOServiceItem){
+	// 	oldServiceItem := testdatagen.MakeDefaultMTOServiceItem(suite.DB())
+	// 	oldServiceItemPrime := testdatagen.MakeMTOServiceItem(suite.DB(), testdatagen.Assertions{
+	// 		Move: testdatagen.MakeAvailableMove(suite.DB()),
+	// 	})
+	// 	return oldServiceItem, oldServiceItemPrime
+
+	// }
```
Comment (sandy-wright) [inline pkg/services/mto_service_item/mto_service_item_updater_test.go:141]: Can this be deleted?

File: pkg/services/mto_service_item/mto_service_item_validators_test.go [hunk 2]
```diff
@@ -230,34 +284,56 @@ func (suite *MTOServiceItemServiceSuite) TestUpdateMTOServiceItemData() {
 	})
 
 	// Test getVerrs for unsuccessful example
-	suite.T().Run("getVerrs - failure", func(t *testing.T) {
+	suite.Run("getVerrs - failure", func() {
+		// Under test:  getVerrs returns a list of validation errors
+		// Set up:      Create a service item, edit the non-prime fields and linked ids
+		//              Run 2 validations that should fail
+		// Expected outcome: No errors
```
Comment (sandy-wright) [inline pkg/services/mto_service_item/mto_service_item_validators_test.go:291]: We do expect some errors from this one - an InvalidInputError and some validation errors

**Review by sandy-wright — APPROVED** [review]
LGTM! Couple of small comments, but this is great. You added a lot of really good documentation to these tests, whew! Made me appreciate the fact that the test logic for the handlers is much more straight-forward and repetitive than the service items haha

File: pkg/services/mto_service_item/mto_service_item_updater_test.go (hunk 1 above)
Comment (shimonacarvalho) [reply pkg/services/mto_service_item/mto_service_item_updater_test.go]: yup

File: pkg/services/mto_service_item/mto_service_item_validators_test.go (hunk 2 above)
Comment (shimonacarvalho) [reply pkg/services/mto_service_item/mto_service_item_validators_test.go]: fixed!

**Pull request merged**